components/FormInput/SharedFormInput.js

```diff
@@ -60,19 +60,23 @@ export default class SharedFormInput extends Component {
       error,
       value,
       label,
+      labelOutside,
       ...remainingProps,
     } = this.props;
+
     const { focus, id } = this.state;
 
-    const active = value && !!value.length;
-    const isInsideVariant = !!placeholder;
+    const hasValue = !!value && value.length;
+    const hasPlaceholderAndValue = !!placeholder && hasValue;
+    const labelIsOutside = labelOutside || hasPlaceholderAndValue;
+    const labelIsOutsideWithPlaceholder = labelOutside && !!placeholder;
 
     const outerCSS = [
       css.container,
       required ? css.required : null,
-      isInsideVariant ? css.labelInside : css.labelOutside,
+      labelIsOutside ? css.labelOutside : css.labelInside,
+      labelIsOutsideWithPlaceholder ? css.labelOutsideWithPlaceholder : null,
       error ? css.error : null,
-      active ? css.active : null,
```

Comment:
This doesn't make sense...

Previously, if the input has a placeholder the input is the inline-label variant.
If not, the label is in place of the placeholder until `active` is true. `active` is only true if a valid value is entered.

So in the CSS: if the input was `active` or was focused on (and is an outside-label variant), the label is moved outside.

With the new behaviour, the label is only moved outside when there's a placeholder and it's an outside-variant.

Inputs that don't have a placeholder now won't work like the used to.

# tl;dr preview

![inputs](https://cloud.githubusercontent.com/assets/1926464/13496301/55d8176e-e147-11e5-963b-b56be11e6595.gif)

![screenshot 2016-03-03 13 53 33](https://cloud.githubusercontent.com/assets/1926464/13496304/58e1e322-e147-11e5-8805-112d3115d18b.png)